server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -6251,9 +6267,111 @@ public UserVm createVirtualMachine(DeployVMCmd cmd) throws InsufficientCapacityE
                 }
             }
         }
+
+        applyLeaseOnCreateInstance(vm, leaseDuration, leaseExpiryAction, svcOffering);
         return vm;
     }
 
+    protected void validateLeaseProperties(Long leaseDuration, String leaseExpiryAction) {
+        if (!VMLeaseManagerImpl.InstanceLeaseEnabled.value()
+                || (leaseDuration == null && StringUtils.isEmpty(leaseExpiryAction))) { // if both are null
```

Comment:
You could also add ` || (leaseDuration != null && leaseDuration < 1)` and reduce the if block below